python/pyarrow/gandiva.pyx

```diff
@@ -150,10 +163,49 @@ cdef class Filter:
             batch.sp_batch.get()[0], selection))
         return SelectionVector.create(selection)
 
+
+cdef inline DataType _ensure_type(object type):
```

Comment:
There is an `_as_type` function defined in `types.pxi`

Reply:
Thanks, I'm using that now (but it requires to make that symbol available in lib.pxd, not sure if we want that).

Reply:
Converted it to a pure python function. We have another `cdef inline _ensure_type` in array.pxi which causes other issues, will resolve it in a follow-up PR, see https://issues.apache.org/jira/browse/ARROW-3735